src/go/k8s/apis/redpanda/v1alpha1/cluster_webhook.go

```diff
@@ -599,10 +607,16 @@ func (r *Cluster) checkCollidingPorts() field.ErrorList {
 	}
 
 	if internal := r.InternalListener(); internal != nil {
+		externalListener := r.ExternalListener()
```

Comment:
nit: could you extract the port collection in a separate method? Also, this method could populate the ports that don't have value.

Something like:
```
func getAllPort() []listenerPorts { ... }
```

Reply:
👍 done